src/stepper/Step.js

```diff
@@ -0,0 +1,126 @@
+import PropTypes from 'prop-types';
+
+import {
+  StepStyles,
+  StepButtonStyles,
+  StepCircleStyles,
+  StyledStepLabelWrapper,
+  StepDescriptionStyles,
+  StepIconStyles,
+  LineStyles,
+  StyledStepText
+} from './Stepper.styles';
+
+const Step = (props) => {
+  const {
+    number,
+    description,
+    icon,
+    activeClassName,
+    className,
+    component,
+    onClick,
+    href,
+    dataTestId,
+    ...rest
+  } = props;
+
+  let as = 'div';
+  if (href) {
+    as = 'a';
+  } else if (!href && onClick) {
+    as = 'button';
+  }
+
+  let customComponentProps = {};
+  if (as === 'button') {
+    customComponentProps = { type: 'button', ...rest };
+  }
+  if (as === 'a') {
+    customComponentProps = { ...rest };
+  }
+  if (component) {
+    as = component;
+    customComponentProps = { ...rest };
+  }
+
+  return (
+    <StepStyles>
+      <StepButtonStyles
+        as={as}
+        onClick={onClick}
+        href={href}
+        $activeClassName={activeClassName}
+        className={className}
+        data-testid={dataTestId}
+        {...customComponentProps} // eslint-disable-line
+      >
+        <StepCircleStyles>
+          <StepIconStyles icon={icon} />
+        </StepCircleStyles>
+        <StyledStepLabelWrapper>
+          <StyledStepText small>Step {number}</StyledStepText>
+          <StepDescriptionStyles block>{description}</StepDescriptionStyles>
+        </StyledStepLabelWrapper>
+      </StepButtonStyles>
+      <LineStyles />
+    </StepStyles>
+  );
+};
+
+Step.defaultProps = {
+  number: null,
```

Comment:
Thoughts about naming this `step` instead of `number`? Since its a step component?